torch/csrc/jit/passes/onnx/scalar_type_analysis.cpp

```diff
@@ -271,6 +285,42 @@ static void UpdateScalarTypeForOutput(
       CreateProfiledTensorTypeWithScalarType(output_tensor_type, scalar_type));
 }
 
+static void RecoverScalarTypeForOutput(
+    Node* n,
+    const c10::ScalarType& scalar_type) {
+  const int64_t onnx_type = ScalarTypeToONNXType(scalar_type);
+  Node* cast_node = n->owningGraph()->create(onnx::Cast, 1);
+  cast_node->addInput(n->output());
+  cast_node->i_(attr::to, onnx_type);
+  cast_node->insertAfter(n);
+  n->output()->replaceAllUsesAfterNodeWith(cast_node, cast_node->output());
+}
+
+static void LowPrecisionCastForStandardOps(Node* n, int opset_version) {
+  TORCH_INTERNAL_ASSERT(n->inputs().size() == 2)
```

Comment:
i think onnx::Gemm could potentially have 3 inputs....

Reply:
Added the Loop to decide to the output and inputs node type.